crates/wit/src/token_stream_generator/record_generator/record_deserializer.rs

```diff
@@ -103,7 +103,7 @@ impl RecordDeserializerGlueCodeGenerator for fce_ast_types::AstRecordItem {
                         let #field = unsafe { String::from_raw_parts(raw_record[#ptr_id] as _, raw_record[#size_id] as _, raw_record[#size_id] as _) };
                     }
                 }
-                ParsedType::Vector(ty) => {
+                ParsedType::Vector(ty, _) => {
```

Comment:
could you please add a comment on the meaning of the following?
```

                    let ptr_id = value_id;
                    let size_id = value_id + 1;
                    value_id += 1;
```
I'm kinda confused on why `value_id` is incremented here, but not for primitive types like `F32`.